src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -143,11 +144,14 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
         this.partialConnectionString.password = password;
     }
 
-    public supportsStoredProcedures(): boolean {
+    public async supportsStoredProcedures(clientConfig: ClientConfig): Promise<boolean> {
         // `semver.gte` complains when a version doesn't have decimals (i.e. "10"), so attempt to convert version to SemVer
         let version: SemVer | null;
         if (this._serverVersion) {
             version = coerce(this._serverVersion);
+        } else {
+            const result = await runPostgresQuery(clientConfig, `SHOW server_version;`);
```

Comment:
It's inefficient to run this query so often when the result should never change - just cache the result inside `this._serverVersion`